### PR TITLE
Extracting Errors

### PR DESCRIPTION
**Pull request by bobgodwinx**
## Flatten out error
Hi all I hope this should close [PR issue 168](https://github.com/RxSwiftCommunity/Action/pull/163) So I went for a KISS solution. 

- [x] Added extension `underlyingError` in order to get `Swift.Error` 
- [x] Added specs to cover claims

### CODE REVIEW

**Comment by freak4pc** [comment]
CI is seemingly failing

File: Tests/ActionTests/ActionTests.swift [hunk 1]
```diff
@@ -50,6 +50,7 @@ class ActionTests: QuickSpec {
 			var enabled: TestableObserver<Bool>!
 			var executing: TestableObserver<Bool>!
 			var executionObservables: TestableObserver<Observable<String>>!
+            var underlyingError: TestableObserver<Error>!
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: indent

File: Tests/ActionTests/ActionTests.swift [hunk 2]
```diff
@@ -58,6 +59,7 @@ class ActionTests: QuickSpec {
 				enabled = scheduler.createObserver(Bool.self)
 				executing = scheduler.createObserver(Bool.self)
 				executionObservables = scheduler.createObserver(Observable<String>.self)
+                underlyingError = scheduler.createObserver(Error.self)
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: indent

**Comment by bobgodwinx** [comment]
@freak4pc  all good now and it can now go into master. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Changelog.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ Changelog
 
 Current master
 --------------
+3.11.0
+-------
+- Introduction of  `underlyingError` streams 🔥 which returns a `Swift.Error` type.
```
Comment (freak4pc) [inline Changelog.md]: What's with the flame Emoji? :)

File: Sources/Action/Action+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
+//
+//  Action+Extensions.swift
+//  Action
+//
+//  Created by Obi Bob on 01.03.19.
+//  Copyright © 2019 CezaryKopacz. All rights reserved.
```
Comment (freak4pc) [inline Sources/Action/Action+Extensions.swift]:
```suggestion
//  Copyright © 2019 RxSwift Community. All rights reserved.
```

File: Sources/Action/Action+Extensions.swift [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        return errors.flatMap { actionError -> Observable<Error> in
+            guard case .underlyingError(let error) = actionError else {
+                return Observable.empty()
+            }
+            return Observable.just(error)
+        }
+    }
+}
+
+extension Action where Input == Void {
```
Comment (freak4pc) [inline Sources/Action/Action+Extensions.swift]:
```suggestion
public extension Action where Input == Void {
```

File: Sources/Action/Action+Extensions.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+//  Action
+//
+//  Created by Obi Bob on 01.03.19.
+//  Copyright © 2019 CezaryKopacz. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+extension Action {
```
Comment (freak4pc) [inline Sources/Action/Action+Extensions.swift]: I think the extensions has to be public or consumers won't be able to use it

```suggestion
public extension Action {
```

File: Tests/ActionTests/ActionTests.swift [hunk 7]
```diff
@@ -92,6 +98,7 @@ class ActionTests: QuickSpec {
 				action.enabled.subscribe().disposed(by: disposeBag)
 				action.executing.subscribe().disposed(by: disposeBag)
 				action.executionObservables.subscribe().disposed(by: disposeBag)
+                action.underlyingError.subscribe().disposed(by: disposeBag)
```
Comment (freak4pc) [inline Tests/ActionTests/ActionTests.swift]: indent
```suggestion
               action.underlyingError.subscribe().disposed(by: disposeBag)
```

**Comment by bobgodwinx** [comment]
@freak4pc final cross check.. I think it's ok now.. we can actually make another release

File: Changelog.md [hunk 8]
```diff
@@ -4,6 +4,10 @@ Changelog
 
 Current master
 --------------
+3.11.0
+-------
+- Introduction of  `underlyingError`  observable which returns a `Swift.Error`  element type.
```
Comment (freak4pc) [inline Changelog.md]:
```suggestion
- Introduction of  `underlyingError` observable which returns a `Swift.Error` element type. 
```

**Pull request merged**